packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/ReactTextPropertiesResolver.kt

```diff
@@ -0,0 +1,133 @@
+package com.datadog.reactnative.sessionreplay
+
+import android.view.Gravity
+import android.widget.TextView
+import com.datadog.android.sessionreplay.model.MobileSegment
+import com.datadog.reactnative.sessionreplay.extensions.convertToDensityNormalized
+import com.datadog.reactnative.sessionreplay.extensions.densityNormalized
+import com.facebook.react.views.text.ReactTextShadowNode
+
+internal class ReactTextPropertiesResolver(
+    private val reactTextShadowNodeUtils: ReactTextShadowNodeUtils
+) {
+    internal fun resolveShadowProperties(
+        view: TextView,
+        pixelDensity: Float,
+        shapeStyle: MobileSegment.ShapeStyle?,
+        border: MobileSegment.ShapeBorder?,
+        textWireframe: MobileSegment.Wireframe.TextWireframe
+    ): MobileSegment.Wireframe {
+
+        val shadowNode = reactTextShadowNodeUtils.getShadowNode(view.id) as? ReactTextShadowNode
+
+        val textStyle = resolveTextStyle(view, pixelDensity, shadowNode)
+        val textPosition = resolveTextPosition(view, pixelDensity, shadowNode)
+
+        return textWireframe.copy(
+                shapeStyle = shapeStyle,
+                border = border,
+                textStyle = textStyle,
+                textPosition = textPosition
+            )
+    }
+
+    private fun resolveTextPosition(
+        view: TextView,
+        pixelsDensity: Float,
+        shadowNode: ReactTextShadowNode?
+    ): MobileSegment.TextPosition {
+        return MobileSegment.TextPosition(
+            resolvePadding(view, pixelsDensity),
+            resolveAlignment(view, shadowNode)
+        )
+    }
+
+    private fun resolveTextStyle(
+        view: TextView,
+        pixelsDensity: Float,
+        shadowNode: ReactTextShadowNode?
+    ): MobileSegment.TextStyle {
+        return MobileSegment.TextStyle(
+            family = reactTextShadowNodeUtils.getFontFamily(view, shadowNode),
+            size = reactTextShadowNodeUtils.getFontSize(view, shadowNode).convertToDensityNormalized(pixelsDensity),
+            color = resolveTextColor(view, shadowNode)
+        )
+    }
+
+    private fun resolvePadding(textView: TextView, pixelsDensity: Float): MobileSegment.Padding {
+        return MobileSegment.Padding(
+            top = textView.totalPaddingTop.densityNormalized(pixelsDensity).toLong(),
+            bottom = textView.totalPaddingBottom.densityNormalized(pixelsDensity).toLong(),
+            left = textView.totalPaddingStart.densityNormalized(pixelsDensity).toLong(),
+            right = textView.totalPaddingEnd.densityNormalized(pixelsDensity).toLong()
+        )
+    }
+
+    private fun resolveAlignment(
+        textView: TextView,
+        shadowNode: ReactTextShadowNode?
+    ): MobileSegment.Alignment {
+        return when (textView.textAlignment) {
+            TextView.TEXT_ALIGNMENT_GRAVITY -> resolveShadowAlignment(textView, shadowNode = shadowNode)
```

Comment:
Can you add comments for each case on how they map to React Native values for alignments?
They can be short but I feel they would make reading this function much easier, and also easier to test our changes.